01_Build_Process/02_Boot_Protocols.md

```diff
@@ -203,115 +203,125 @@ The interface between a higher level language like C and assembly (or another hi
+
+```c
+__attribute__((used, section(".requests")))
+static volatile struct limine_framebuffer_request {
+    .id = LIMINE_FRAMEBUFFER_REQUEST,
+    .revision = 0
 };
 ```
 
-If not familiar with the `__attribute__(())` syntax, it's a compiler extension (both clang and GCC support it) that allows us to do certain things our language wouldn't normally allow. This attribute specified that this variable should go into the `.stivale2hdr` section, as is required by the stivale2 spec.
-
-Next we set some fields in the stivale2 header:
+The requests types are all declared in the `limine.h` header.
 
-- `entry_point`: Is used to override the ELF's entry point address. Set this to zero to use the regular entry function we set in the linker script.
-- `stack`: Self explanatory, used to set the stack the kernel code will start with.
-- `flags`: A bitfield of flags. Bit 1 asks the bootloader to return higher half addresses to us for tags, modules and other things. Bit 2 asked the bootloader to make use of the nx-bit and write-enable bits in the page tables when loading the kernel. Bit 3 is recommended and enables the bootloader to load us at any physical address as long as the virtual address is the same. Bit 4 is required to be set, as it disables a legacy feature.
-- `tags`: A pointer to the first stivale2 tag in the linked list of requests.
+Any type of information that needs to be requested to the bootloader, will have it's own `request` variable, with it's type.
 
-In the example above we actually set the first tag to a framebuffer request, so lets see what that would look like:
+Finally the requests start and end markers needs to be declared:
 
 ```c
-static stivale2_header_tag_framebuffer framebuffer_tag =
-{
-    .tag =
-    {
-        .identifier = STIVALE2_HEADER_TAG_FRAMEBUFFER,
-        .next = 0,
-    },
-    .framebuffer_width = 0,
-    .framebuffer_height = 0,
-    .framebuffer_bpp = 0
-};
+__attribute__((used, section(".requests_start_marker")))
+static volatile LIMINE_REQUESTS_START_MARKER;
+
+__attribute__((used, section(".requests_end_marker")))
+static volatile LIMINE_REQUESTS_END_MARKER;
 ```
 
-The `framebuffer_*` fields can be used to ask for a specific kind of framebuffer, but leaving them to zero tells the bootloader we want to best possible available. The `next` field can be used to point to the next header tag, if we had another one we wanted. The full list of tags is available in the stivale2 specification (see the useful links appendix).
+Again they can be placed anywhere in the code, since their position will be decided by the section they belongs to.
 
-The last detail is to change the signature of our kernel entry function to:
+The last detail is to add the kernel start function (declared in the `ENTRY()` section in the linker script):
```

Comment:
since you mention below that the entry function doesnt require any particular signature - I think we can remove this sentence and the code snippet below?